src/client.ts

```diff
@@ -1492,15 +1500,24 @@ export class Client {
 	 */
 	private async getResolvedRefString(params?: FetchParams): Promise<string> {
 		if (this.refState.autoPreviewsEnabled) {
-			let previewRef: string | undefined = undefined;
+			let previewRef: string | undefined;
+
+			let cookieJar: string | null | undefined;
 
 			if (globalThis.document?.cookie) {
-				previewRef = getCookie(cookie.preview, globalThis.document.cookie);
-			} else if (this.refState.httpRequest?.headers?.cookie) {
-				previewRef = getCookie(
-					cookie.preview,
-					this.refState.httpRequest.headers.cookie,
-				);
+				cookieJar = globalThis.document.cookie;
+			} else if (this.refState.httpRequest?.headers) {
+				if ("get" in this.refState.httpRequest?.headers) {
```

Comment:
Shouldn't we check with `instanceof Headers` or `typeof === "function"` instead? This could fail if there's a "get" header 🤔

Reply:
Yep, you're right. Will fix! 🙂

Reply:
I went with an extra `typeof === "function"` since `Headers` doesn't exist in Node. Duck-typing is simpler in this case.

Reply:
Awesome!